### PR TITLE
RD2: Consolidate Payment Widget State into App State object

### PR DESCRIPTION
**Pull request by bdvorachek**
WI: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000cEQzOYAW/view
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: force-app/main/default/lwc/rd2EntryForm/rd2EntryForm.js [hunk 1]
```diff
@@ -762,27 +678,23 @@ export default class rd2EntryForm extends LightningElement {
...
-     * @description Handle component display when an error occurs
-     * @param isDisabled: Indicates if the Save button should be disabled
-     */
-    setSaveButtonDisabled(isDisabled) {
-        const disableBtn = !!(isDisabled && isDisabled === true);
+    disableSaveButton() {
+        this.isSaveButtonDisabled = true;
+    }
 
-        this.isLoading = disableBtn;
```
Comment (lparrott) [inline force-app/main/default/lwc/rd2EntryForm/rd2EntryForm.js:784]: I think this change is preventing error messages from appearing, the Loading spinner stays visible, hiding the error.

Comment (bdvorachek) [reply force-app/main/default/lwc/rd2EntryForm/rd2EntryForm.js:784]: I think we may need to add a line to `handleError` to disable the loading spinner to fully replicate the former behavior of `setSaveButtonDisabled`. Will see if that fix works and get a test in there for it if so.

**Review by dfuller114 — APPROVED** [review]
great work Ben!

File: force-app/main/default/classes/RD2_EntryFormController.cls [hunk 2]
```diff
@@ -188,8 +224,8 @@ public with sharing class RD2_EntryFormController {
      * false, the custom UI will display a warning message at the top of the page.
      * @return True if the User has Read access to all required fields in the UI
      */
-    @AuraEnabled(Cacheable=true)
-    public static Boolean hasRequiredFieldPermissions() {
+    @TestVisible
+    private static Boolean hasRequiredFieldPermissions() {
```
[the remark this answers is not in the document]
Comment (bdvorachek) [reply force-app/main/default/classes/RD2_EntryFormController.cls:228]: Thanks, changed this over!

**Pull request merged**